aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncType.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.syncengine;
+
+/**
+ * Enum representing the type of initial sync.
+ */
+public enum SyncType {
```

Comment:
Ahh. Hm, yea.

So, this looks reasonable. Can you create a separate isolated PR for this?

If we treat this seriously, I think we can make use of `SyncType` in a number of places where delta vs. base sync logic is managed currently. If I'm not mistaken, it is a nullability check in most places right now. Like, do I have a timestamp yes vs. no? However, it would be cleaner to be like:
```java
if (lastSyncTime == null) {
    syncType = SyncType.FULL;
}
```
etc.

TLDR there's a bunch of places you can use this, and it'll probably be significant enough for its own (~low complexity, but potentially verbose) PR.

Reply:
Just posted #713 

Note that in that PR, I'm using `BASE` instead of `FULL` since it appears to be more consistent with what I've seen in the docs. 

I have some other changes to this PR which should address some of the other issues. Once we get #713 merged, I can update this PR.